packages/mobx-state-tree/src/types/utility-types/reference.ts

```diff
@@ -201,7 +202,7 @@ export interface ReferenceOptions<T> {
     set(value: T, parent: IAnyStateTreeNode | null): string | number
 }
 
-export interface IReferenceType<IR extends IAnyModelType>
+export interface IReferenceType<IR extends IAnyComplexType>
```

Comment:
had to change this to complex type since now the union of models is a complex type (which I guess makes more sense since you can't use props, actions etc over a union of models)

Reply:
Should be fine I think